### DISCUSSION TITLE
Using text files as database

### DISCUSSION THREAD

**Discussion by anjanesh** [vercel/next.js#39392] [Help] [answered]
Is there a way for NextJS to read a markdown textfile from a a folder called, say for example, **/md-data/about.md** and include it into a pages file, say **pages/about.tsx**. (Client Side only)

I want to do this so that a non-developer can edit the about.md file with the content he has in mind.

I intend to `next export` and not run Next on the server side via nodeJS. I am going to deploy this as a pure client side static site.

**Comment by alex289** [comment, marked as answer]
Yes that is possible. To make this work you have to use fs inside GetStaticProps.

This [example](https://github.com/vercel/next.js/tree/canary/examples/blog-starter) should be kind of what you want.

**Reply by anjanesh** [reply]
I checked - according to this - https://github.com/vercel/next.js/blob/canary/examples/blog-starter/lib/api.ts - it has `const fileContents = fs.readFileSync(fullPath, 'utf8')` - how exactly will this work on browser client-side JavaScript ?

**Reply by alex289** [reply]
The file you are referring to is used in GetStaticProps while building the app to static html. It is not possible to use fs on the client. If you want to update the file and use it after build you need to use GetServerSideProps I think but this would make the deployment kinda difficult. I'd prefer to use GetStaticProps and make a new deployment when changing a document
